hypothesis-python/tests/cover/test_exceptiongroup.py

```diff
@@ -0,0 +1,154 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Copyright the Hypothesis Authors.
+# Individual contributors are listed in AUTHORS.rst and the git log.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+import pytest
+
+from hypothesis import errors, given, strategies as st
+from hypothesis.internal.compat import BaseExceptionGroup, ExceptionGroup
+from hypothesis.strategies import DataObject
+
+
+def test_discard_frozen() -> None:
+    @given(st.data())
+    def discard_frozen(data: DataObject) -> None:
+        # raising Frozen doesn't actually do anything, what matters is
+        # whether the data is frozen.
+        data.conjecture_data.freeze()
+        raise ExceptionGroup("", [errors.Frozen()])
```

Comment:
I think calling `.freeze()` like this is technically an invalid use of an internal API; that seems OK for the test but it'd be good to say so in a comment to avoid future confusion.

Reply:
oh interesting. I thought if I stuck to non-underscored methods everywhere I'd be using public APIs, maybe something should be (deprecated and) renamed to make it clearer for users?